temporalio/converter.py

```diff
@@ -1209,6 +1279,50 @@ def decode_search_attributes(
     return ret
 
 
+def decode_typed_search_attributes(
+    api: temporalio.api.common.v1.SearchAttributes,
+) -> temporalio.common.TypedSearchAttributes:
+    """Decode API search attributes to typed search attributes.
+
+    Args:
+        api: API message with search attribute values to convert.
+
+    Returns:
+        Typed search attribute collection.
+    """
+    conv = default().payload_converter
+    pairs: List[temporalio.common.SearchAttributePair] = []
+    for k, v in api.indexed_fields.items():
+        # We want the "type" metadata, but if it is not present or an unknown
+        # type, we will just ignore
+        key = temporalio.common.SearchAttributeKey._from_metadata_type(
+            k, v.metadata.get("type").decode()
+        )
+        if key:
+            val = conv.from_payload(v)
+            # If the value is a list but the type is not keyword list, pull out
+            # single item or consider this an invalid value and ignore
+            if (
+                key.indexed_value_type
+                != temporalio.common.SearchAttributeIndexedValueType.KEYWORD_LIST
+                and isinstance(val, list)
+            ):
+                if len(val) != 1:
+                    continue
```

Comment:
Why are we skipping this?
Is this behavior consistent with Java (too lazy to check myself).

Reply:
I believe so. I was told this should never happen with newer ones and we want to be strict here. I will set a reminder to confirm w/ Java behavior.

Reply:
Same in Java. For the strict interface if we get a list for non-keyword-list and it isn't a single value, server contract is wrong and we ignore.